### ISSUE TITLE
[BUG]what‘s is that？I'm getting an error.

### ISSUE THREAD

**Issue by Chengxming** [charlesbel/Microsoft-Rewards-Farmer#160]
<img width="873" alt="屏幕截图_20230225_172924" src="https://user-images.githubusercontent.com/114639900/221349973-e7bb1567-adec-473f-9218-19b8e23ff9a2.png">
I am sorry,I don't know what's the error'

**Comment by x68000x** [comment]
This happens in both a standard installation of Windows (10) or (Ubuntu/CentOS) Linux. 

It does not happen on Mac OS X(13.2.1) and it does not happen on a Raspberry Pi 3B with the current Raspberry Pi OS

**Comment by Chengxming** [comment]
What can I do for it.

**Comment by AlexyGalaxy** [comment]
Make sure to manually log in to your bing account, go to https://rewards.bing.com/ and click "Join rewards program" or "Join Bing Rewards" The script will not do that for you.

**Comment by Chengxming** [comment]
I have joined rewards when using the script, and I also have some points.

**Comment by AlexyGalaxy** [comment]
You dont seem to be connecting there might be an issue with the chrome driver you installed. try to download an older version 

-Install ChromeDriver from https://chromedriver.chromium.org/downloads
	Place the file in X:\Windows (X as your Windows disk letter)

**Comment by Chengxming** [comment]
Thank you for your help, I will try it tomorrow, thank you very much. @AlexyGalaxy 😃

**Comment by Chengxming** [comment]
I tried but encountered a new problem
![%~R P`(K YFSV4KLAL}FTP5](https://user-images.githubusercontent.com/114639900/221743050-2bb32a90-df68-499e-8d56-bc758eefb048.png)

**Comment by Chengxming** [comment]
@AlexyGalaxy

**Comment by AlexyGalaxy** [comment]
What region are you in? US or?

**Comment by Chengxming** [comment]
I used Authentic passwordless login Microsoft so there was a problem, I will turn off passwordless login Microsoft and then try again, thank you for your help!

**Comment by AlexyGalaxy** [comment]
It sounds like you never got the script fully running because it is looking for a particular element that is used in you region. Are you outside of the US?

Try changing waitUntilVisible to the code below and let me know what happens.

waitUntilVisible(browser, By.XPATH, '//*[@id="navs"]/div/div/div/div/div[4]/div/button/span/i', 20)
Also disable headless mode by commenting out line 46 and 47 with #. Then watch the script open chrome and look to see where it is stuck. (When done testing make sure to delete the # to rentable headless mode.

#if headless_mode : #comment out to disable headless mode (makes window visable) 
           #options.add_argument("--headless") #comment out to disable headless mode (makes window visable) 
Pause the script by placing the code located below, before waitUntilVisible and once again after waitUntilVisible. Then run the script and post the screen shot here. If the browser is taking too long to load, increase the number at the end of waitUntilVisible to 30-50(this is the time waited)

print('[TEST] Press any key to continue')
input()
If it doesnt work read this:
The way this line of code works is by grabbing the XPATH information. When your script is paused and waiting for your second input to continue, click on the browsers that was opened and running the script, press F12(this will open a page showing the browsers code), then press ctrl+shift+c, then hover over the top banner of the website and right click. Now in that new page that shows the browsers code there is a line highlighted. Right click that line and select Copy>Copy XPATH. Paste that line of code within waitUntilVisible.

waitUntilVisible(browser, By.XPATH, 'PASTE_XPATH_CODE_HERE/DELETE_THESE_WORDS_WITHIN_THE_QUOTES', 20)
let me know if that helps. If so please commnet below with the changes you made to the script.

Someone had this exact issue but unfortunately they did not post the exact changes they made.
https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/152

**Comment by Chengxming** [comment]
When I cancel the passwordless login, I log in successfully but it seems that it cannot run normally.
![D$ZEAYGOCB{{@GOF(670KBD](https://user-images.githubusercontent.com/114639900/221773567-bf43b65d-618a-477c-b287-d2aee19945f5.png)

**Comment by Chengxming** [comment]
@AlexyGalaxy I disabled headless mode and ran the script and found an error, and got stuck on the following interface.
![NKLNQR4ZA UH4QZAROTAR%K](https://user-images.githubusercontent.com/114639900/221808066-89ed6114-76c0-4d21-a257-75c131bfda77.png)
![WVUK9EZIB~JZ_P1F~0{@K3H](https://user-images.githubusercontent.com/114639900/221808086-916090cc-a43a-47cc-b38e-765517382356.png)

**Comment by Chengxming** [comment]
@AlexyGalaxy   I disabled headless mode and ran the script and found an error, and got stuck on the following interface.
![image](https://user-images.githubusercontent.com/114639900/221808584-ccdd68d0-b16a-4f10-8811-f1a54dbaf055.png)
![image](https://user-images.githubusercontent.com/114639900/221808729-4d7c793a-6ef5-4d7b-8aac-25b2a1599ced.png)

**Comment by AlexyGalaxy** [comment]
You are not in the US so i cant be of much help...
Try to pause the script by placing the code located below, before waitUntilVisible and once again after waitUntilVisible. Then run the script in headless mode and wait for the page.
```
print('[TEST] Press any key to continue')
input()
```

Once the script is pasues and waiting for an input you need to click on the browser that was opened and is running the script, press F12(this will open a page showing the browsers html code), then press ctrl+shift+c, then hover over the top banner of the website and right click on Rewards. Now in that new page that shows the browsers code there is a line highlighted. Right click that line and select Copy>Copy XPATH. Paste that line of code within waitUntilVisible.

![image](https://user-images.githubusercontent.com/86795638/221856219-53b307c1-aadf-4859-b56e-5bdce420e9cb.png)

![image](https://user-images.githubusercontent.com/86795638/221856646-ec2c6ab7-7a05-4f94-9dec-6d35c497d7b3.png)

```
waitUntilVisible(browser, By.XPATH, 'PASTE_XPATH_CODE_HERE/DELETE_THESE_WORDS_WITHIN_THE_QUOTES', 20)
```

This part of the script is trying to open the rewards dashboard. If this doesnt fix the issue you next solution is to manually direct the script using the code below insdead of waitUntilVisible. 
```
time.sleep(3)
browser.get('ENTERURLHERE.com')
time.sleep(3)
```
Like i said there have been 2 other people with this issue and unfortunately they did not post their solution. if you figure it out please post what fixes this issue so i may work it into the code. 
https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/154
https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/152

**Comment by Chengxming** [comment]
I am a beginner in python, your code is very difficult for me, I’m sorry I can’t find out what’s wrong with it. I reverted all the changes and turned off headless mode, then tried running it several times and it seemed to work once. Here is the successful content.
![GIF 2023-3-1 2-06-58](https://user-images.githubusercontent.com/114639900/221943055-fd9067a8-af8c-4440-afe0-090935f3657b.gif)

**Comment by AlexyGalaxy** [comment]
seems like it did all searches as intended. I cannot troubleshoot your issues bc I am in the US and you are in Asia. I told you the steps i take to edit XPATH code when bing rewards changes their site there is not much more i can do. sorry

**Comment by Imgkl** [comment]
@AlexyGalaxy I'm facing this issue too and I'm in India. I think what's happening is, On Asia/India Microsoft acounts page, we don't have the rewards tab. Since because of this, the browser is not able to find the element.

US:
<img width="810" alt="image" src="https://user-images.githubusercontent.com/25414681/222081262-757b7fb8-3791-4d93-9037-7cf60fee74bd.png">
IN: 
<img width="1800" alt="image" src="https://user-images.githubusercontent.com/25414681/222081391-5eb24bbb-31bb-4c60-afd1-9c333bfea5c2.png">

**Comment by AlexyGalaxy** [comment]
when you go to this url does it bring you to the rewards dashboard? https://rewards.microsoft.com

**Comment by Imgkl** [comment]
@AlexyGalaxy Yes.

**Comment by AlexyGalaxy** [comment]
I am having lots of issues with the script today. Seems bing is updating/updated their site.

During log in I am sometimes asked to click my account and reEnter my password. Then both of those links to the rewards dashboard are no longer working/take too long. 

I will wait until tomorrow to see if things start to work again or I can figure out a solution.

**Comment by AlexyGalaxy** [comment]
@Imgkl So both links bring you to the rewards dashboard?

**Comment by Imgkl** [comment]
Yes @AlexyGalaxy, Both links takes me to the rewards dashboard.

**Comment by AlexyGalaxy** [comment]
Try something for me... Put your script into headless mode, then delete the waitUntilVisable function and the if else statement after and paste the code below:
```
time.sleep(random.randint(10, 15))

try :
    prYellow('[TEST] Loading https://rewards.bing.com/') #test
    BASE_URL = 'https://rewards.bing.com/'
    browser.get(BASE_URL)
except:
    prRed('[ERROR] Could not Loading https://rewards.bing.com/')#test
    pass
```
It should look like this.
![image](https://user-images.githubusercontent.com/86795638/222167285-1a305215-3700-48cd-a01a-b76864c69ae5.png)

Run this with headless mode disabled and let me know the outcome.

**Comment by Imgkl** [comment]
Getting 504 when I try to access through the script. but when opened manually it shows the dashboard.

**Comment by AlexyGalaxy** [comment]
Welp this is not good. Not sure about a solution. Maybe they are updating part of their network... or they figured out how to stop selenium users from accessing their site. I will wait until tomrrow to try again.

 If you get something working please post it here.

**Comment by Imgkl** [comment]
weird. In the same selenium window, I was able to open rewards.bing.com without any issues. 

Sure, If I find anything useful, I'll ping here.

**Comment by AlexyGalaxy** [comment]
The issue no longer occurs. It must have been an issue with bing. The code i provided you does work instead of waitUntilVisable function

**Comment by CPTHARLOCK96** [comment]
> I am a beginner in python, your code is very difficult for me, I’m sorry I can’t find out what’s wrong with it. I reverted all the changes and turned off headless mode, then tried running it several times and it seemed to work once. Here is the successful content. ![GIF 2023-3-1 2-06-58](https://user-images.githubusercontent.com/114639900/221943055-fd9067a8-af8c-4440-afe0-090935f3657b.gif) [ ![GIF 2023-3-1 2-06-58](https://user-images.githubusercontent.com/114639900/221943055-fd9067a8-af8c-4440-afe0-090935f3657b.gif) ](https://user-images.githubusercontent.com/114639900/221943055-fd9067a8-af8c-4440-afe0-090935f3657b.gif) [ ](https://user-images.githubusercontent.com/114639900/221943055-fd9067a8-af8c-4440-afe0-090935f3657b.gif)

where i can find the 3.0 version?

**Comment by AlexyGalaxy** [comment]
Updated verison is here on my github. Make sure to set up the Logger and errLoger (read the README) https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

**Comment by charlesbel** [comment]
Fixed with v3.0 #188
